pkg/destroy/openstack/openstack.go

```diff
@@ -653,6 +653,57 @@ func getRouterByPort(client *gophercloud.ServiceClient, allPorts []ports.Port) (
 	return "", nil
 }
 
+func deleteLeftoverLoadBalancers(opts *clientconfig.ClientOpts, logger logrus.FieldLogger, networkID string) (bool, error) {
+	conn, err := clientconfig.NewServiceClient("load-balancer", opts)
+	if err != nil {
+		// Ignore the error if Octavia is not available for the cloud
+		var gerr *gophercloud.ErrEndpointNotFound
+		if errors.As(err, &gerr) {
+			logger.Debug("Skip load balancer deletion because Octavia endpoint is not found")
+			return true, nil
+		}
+		logger.Error(err)
+		return false, nil
+	}
+
+	listOpts := loadbalancers.ListOpts{
+		VipNetworkID: networkID,
+	}
+	allPages, err := loadbalancers.List(conn, listOpts).AllPages()
+	if err != nil {
```

Comment:
I suggest to handle ErrDefault403 errors separately  to prevent situations when a user doesn't have permissions to list load balancers

Reply:
Hmm, this is a pattern we're using all over the place in this file, and if we wanted to treat 403 differently I think we should do it in a separate change.
How do you suggest we should handle a ErrDefault403 ?

Reply:
```
if err != nil {
    var gerr gophercloud.ErrDefault403
    if !errors.As(err, &gerr) {
        logger.Debugf("It's forbidden to list load balancers")
        return true, nil
    }
    logger.Error(err)
    return false, nil
}
```